### PR TITLE
Added annotatealleles command

### PR DESCRIPTION
**Pull request by lfrancioli**
- added option to keep * alleles in SplitMulti
- changed va.aIndex from 0 to 1 for bi-allelic sites (consistent with multi-allelic sites) in SplitMulti

### CODE REVIEW

File: docs/commands/annotatealleles_expr.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Notes:
+
+This command runs similarly to [annotatevariants expr](commands.html#annotatevariants_expr) but it dynamically splits multi-allelic sites,
+ computes each expressions on each split allele separately, and returns an **Array** with one entry per
+   non-ref allele for each expression.
+
+During the evaluation of the expressions, two additional variant annotations are accessible:
+1. `va._aIndex`: An **Int** indicating the index of the non-reference allele being evaluated
+(amongst all alleles including the reference allele, so cannot be `0` )
+2. `va._wasSplit`: A **Boolean** indicating whether this allele belongs to a multi-allelic site or not.
```
Comment (cseed) [inline docs/commands/annotatealleles_expr.md]: I don't see why and underscore is necessary for these annotations.  In particular, it seems nice for an expression E to work both in:

  annotateallleles expr -c E

and

  splitmulti
  annotatevariants expr -c E

File: python/pyhail/dataset.py [hunk 2]
```diff
@@ -42,6 +42,20 @@ def aggregate_intervals(self, input, condition, output):
         pargs = ['aggregateintervals', '-i', input, '-c', condition, '-o', output]
         return self.hc.run_command(self, pargs)
 
+    def annotate_alleles_expr(self, condition, propagate_gq=False):
+        """Annotate alleles with expression.
+
+        :param str condition: Annotation expression.
```
Comment (cseed) [inline python/pyhail/dataset.py]: This should be:

```
  :param condition: Annotation expression.
  :type param: str or list of str
```

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateAllelesExpr.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    val cond = options.condition
+    val propagateGQ = options.propagateGQ
+    val isDosage = vds.isDosage
+
+    val (vas2, insertIndex) = vds.vaSignature.insert(TInt, "_aIndex")
+    val (vas3, insertSplit) = vas2.insert(TBoolean, "_wasSplit")
+    val localGlobalAnnotation = vds.globalAnnotation
+
+    val evalva = vds.vaSignature.insert(TInt,"_aIndex")._1.insert(TBoolean,"_wasSplit")._1
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateAllelesExpr.scala]: This is just the same as vas3, right?

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateAllelesExpr.scala [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      "g" -> (3, TGenotype),
+      "s" -> (4, TSample),
+      "sa" -> (5, vds.saSignature))
+    val ec = EvalContext(Map(
+      "global" -> (0, vds.globalSignature),
+      "v" -> (1, TVariant),
+      "va" -> (2, evalva),
+      "gs" -> (3, TAggregable(TGenotype, aggregationST))))
+
+    val (parseTypes, fns) = Parser.parseAnnotationArgs(cond, ec, Some(Annotation.VARIANT_HEAD))
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateAllelesExpr.scala]: You need to rebase.  The Parser got cleaned up.  This is now `parseAnnotationExprs` and the signature has changed slightly.  That's why the test is failing.

**Comment by cseed** [comment]
Extremely nice @lfrancioli, very elegantly done.  Rebase and address the minor comments, and it should be good to go.  A simple test would be nice, too, but I'll put that on our todo list if you don't get to it.

**Comment by lfrancioli** [comment]
Thanks a lot for looking at the PR @cseed! I've made the changes you suggested and added a short test suite. Hope it can go in :)

File: src/test/scala/org/broadinstitute/hail/methods/AnnotateAllelesSuite.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.hail.methods
+
+import org.broadinstitute.hail.SparkSuite
+import org.broadinstitute.hail.driver._
+import org.broadinstitute.hail.io.vcf.LoadVCF
+import org.testng.annotations.Test
+
+/**
```
Comment (tpoterba) [inline src/test/scala/org/broadinstitute/hail/methods/AnnotateAllelesSuite.scala]: delete this comment, I think.  I've turned this behavior off in IntelliJ

**Pull request merged**